pyext/src/analysis_trajectories.py

```diff
@@ -1022,42 +1091,34 @@ def do_extract_models_single_rmf(self, gsms_info, out_rmf_name, traj_dir, analys
         for p in processes:
             p.join()
         
-        output_rmf = analysis_dir + "/" + out_rmf_name
-        scorescat_string = "cat "
+        output_rmf = os.path.join(analysis_dir, out_rmf_name)
+        output_score_file = os.path.join(analysis_dir, scores_prefix+'.txt')
         
-        # Test to see if we have rmf_cat
-        try:
-            subprocess.check_call(["rmf_cat", "-h"], shell=True)
-            rmfcat_string = "rmf_cat "
-            scorescat_string = "cat "
-            # Concatenate the RMF and scorefiles
-            for n in range(len(filenames)):
-                rmfcat_string += filenames[n] +" "
-                scorescat_string += scorefiles[n] +" "
-
-            # output all rmfs to the filename
-            rmfcat_string += output_rmf
-            print("Concatenating",len(filenames)," RMF files to", output_rmf)
+        # use rmf_cat if available
+        if shutil.which('rmf_cat'):
+            # concatenate RMF files
+            rmfcat_string = 'rmf_cat %s %s' % (' '.join(filenames), output_rmf)
+            print('Concatenating', len(filenames), 'RMF files to', output_rmf)
             os.system(rmfcat_string)
-        except subprocess.CalledProcessError:
-            # Otherwise, we need to use RMF
+    
+        # otherwise, fall back on RMF
+        else:
+            print("rmf_cat binary not found on path. Adding frames from output rmf files for each parallel chunk using the RMF library (this will be slow!)")
             orh = RMF.create_rmf_file(output_rmf)
-
-            for n in range(len(filenames)):
-                rh = RMF.open_rmf_file_read_only(filenames[n])
-                scorescat_string += scorefiles[n] +" "
+            for n, fn in enumerate(filenames):
+                rh = RMF.open_rmf_file_read_only(fn)
                 if n==0:
                     RMF.clone_file_info(rh, orh)
                     RMF.clone_hierarchy(rh, orh)
                     RMF.clone_static_frame(rh, orh)
-                orh.set_description(orh.get_description() +"\n" + rh.get_description())
+                orh.set_description(orh.get_description() +'\n' + rh.get_description())
                 for f in rh.get_frames():
                     rh.set_current_frame(f)
                     orh.add_frame(rh.get_name(f), rh.get_type(f))
                     RMF.clone_loaded_frame(rh,orh)
-
+        
         # Concatenate all score files to the same file
-        scorescat_string += "> "+ analysis_dir + "/" + scores_prefix +".txt"
+        scorescat_string = 'cat %s > %s' % (' '.join(scorefiles), output_score_file) 
```

Comment:
Oh no, never do this. `shutil.copyfileobj` is probably your friend here, e.g. (untested):

    with open(output_score_file, 'wb') as outf:
        for scorefile in scorefiles:
            with open(scorefile, 'rb') as scoref:
                 shutil.copyfileobj(scoref, outf)

Reply:
implemented this.